internal/util/update/update.go

```diff
@@ -176,6 +186,23 @@ func (u Command) Run(ctx context.Context) error {
 	return nil
 }
 
+// updateSubKf updates subpackage with given ref and update strategy
+func updateSubKf(subKf *kptfilev1.KptFile, ref string, strategy kptfilev1.UpdateStrategyType) {
+	// check if explicit ref provided
+	if ref != "" {
+		subKf.Upstream.Git.Ref = ref
+	}
+	if strategy != "" {
+		subKf.Upstream.UpdateStrategy = strategy
```

Comment:
why are we updating the strategy?

Reply:
I wanted to maintain parity with what we did via CLI for the root pkg. We swap the strategy if explicitly specified.
https://github.com/GoogleContainerTools/kpt/blob/faf69b23ac6df9740e2c0ee7232ca70f68b70b5f/internal/util/update/update.go#L132-L133

Reply:
Fair enough.